Earthfile

```
@@ -40,7 +40,14 @@ integration-test:
     COPY /integration_test/config/config.exs ./integration_test/config/config.exs
     WORKDIR /src/integration_test
     RUN mix local.hex --force
-    RUN mix deps.get
+
+    # Ensure integration_test/mix.lock contains all of the dependencies we need and none we don't
+    RUN cp mix.lock mix.lock.orig && \
+        mix deps.get && \
+        mix deps.unlock --check-unused && \
+        diff -u mix.lock.orig mix.lock && \
+        rm mix.lock.orig
+
```

```suggestion
    RUN mix deps.unlock --check-unused
```

I removed the comment because it is hopefully self-explanatory. :)

---

I just removed the change from `mix.lock` and only ran `mix deps.unlock --check-unused` on another branch (https://github.com/aaronrenner/phoenix/compare/master...aaronrenner:ar-error-on-mix-lock-changes-try-2) to ensure it fails. Unfortunately [the build](https://github.com/aaronrenner/phoenix/runs/1582511499) passes and does not catch that `phoenix_view` is missing from the mix.lock file.

I'd love to know if I'm overlooking something. It would be nice if there was something like `mix deps.get --error-on-unlocked-dependencies`.

---

Sorry, you are right. You want to check if anything is unused and if anything is missing. My bad. :)